mos-platform/nes-mmc1/mapper.s

```diff
@@ -99,23 +99,45 @@ get_prg_bank:
 .weak set_prg_bank
 __set_prg_bank:
 set_prg_bank:
-	tay
+	bit .Luse_suxrom_bankswitching
+	bpl .Lcontinue_bank_switch
+	; save the new bank byte on the stack for safe keeping
+	pha
+	; check which bits changed
+	eor _PRG_BANK
+	; if the outer bank bit changed then we need to set a new outerbank
+	and #%00010000
+	beq .Lset_inner_bank
+.Lset_new_outer_bank:
+	; Flip the outer bank bit for _CHR_BANK0 and _CHR_BANK1
+	lda _CHR_BANK0
+	eor #%00010000
+	jsr set_chr_bank_0_retry
+	lda _CHR_BANK1
+	eor #%00010000
+	jsr set_chr_bank_1_retry
+.Lset_inner_bank:
+  	; restore the bank byte and put it in y so we can reload it on retry
+  	pla
+.Lcontinue_bank_switch:
+  tay
+  ; original code below
 .Lset:
-	inc __reset_mmc1_byte
-	ldx #1
-	stx _IN_PROGRESS
-	mmc1_register_write MMC1_PRG
-	ldx _IN_PROGRESS
-	beq .Lretry
-	dex
-	stx _IN_PROGRESS
-	sty _PRG_BANK
-	rts
+  inc __reset_mmc1_byte
+  ldx #1
+  stx _IN_PROGRESS
+  mmc1_register_write MMC1_PRG
+  ldx _IN_PROGRESS
+  beq .Lretry
+  dex
+  stx _IN_PROGRESS
+  sty _PRG_BANK
+  rts
 .Lretry:
-	tya
-	jmp .Lset
-
-
+  tya
+  jmp .Lset
+.Luse_suxrom_bankswitching:
+  .byte ((<prg_rom_is_512) << 7)
```

Comment:
This is extremely fishy; there shouldn't be any way that the linker can actually fill in this value. It can't actually read expressions like `(<prg_rom_is_512) << 7` out of the object file for it to resolve them; such expressions can only be resolved by the assembler. But there's no way the assembler can know the value of `prg_rom_is_512`. Does this successfully compile? If so, can you paste the ELF file on Discord? I need to take a look at it; there may be an assembler bug here.